### PR TITLE
feat(data_table): add support to cucumberjs

### LINKED ISSUE #11009: Cucumber.js dataTable type missing.
I tried using the latest `cucumber/cucumber.d.ts` file in this repo and had problems.
  - The authors of that type definition are cc/ @abraaoalves

dataTable type and methods are missing (ex. `.hashes()`)

See https://github.com/cucumber/cucumber-js/blob/master/features/data_tables.feature

### PR DESCRIPTION
**Pull request by AbraaoAlves**
Fix #11009

### CODE REVIEW

**Comment by AbraaoAlves** [comment]
👍

File: cucumber/cucumber.d.ts [hunk 1]
```diff
@@ -10,8 +10,17 @@ declare namespace cucumber {
 		(errror?:any):void;
 	}
 
+  export interface TableDefinition{
```
Comment (horiuchi) [inline cucumber/cucumber.d.ts]: Please unify the indent style with other lines.

File: cucumber/cucumber-tests.ts [hunk 2]
```diff
@@ -1,24 +1,26 @@
 /// <reference path="cucumber.d.ts" />
+/// <reference path="../assert/assert.d.ts" />
 
 function StepSample() {
 	type Callback = cucumber.CallbackStepDefinition;
+  type Table = cucumber.TableDefinition;
```
Comment (vvakame) [inline cucumber/cucumber-tests.ts]: nitpick

File: cucumber/cucumber-tests.ts [hunk 3]
```diff
@@ -36,5 +38,43 @@ function StepSample() {
 			callback(new Error("Expected to be on page with title " + title));
 		}
 	});
+  
+  // Type for data_table.js on
```
Comment (vvakame) [inline cucumber/cucumber-tests.ts]: nitpick

**Comment by horiuchi** [comment]
👍

**Pull request merged**